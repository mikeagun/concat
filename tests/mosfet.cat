#Copyright (C) 2020 D. Michael Agun
#
#Licensed under the Apache License, Version 2.0 (the "License");
#you may not use this file except in compliance with the License.
#You may obtain a copy of the License at
#
#http://www.apache.org/licenses/LICENSE-2.0
#
#Unless required by applicable law or agreed to in writing, software
#distributed under the License is distributed on an "AS IS" BASIS,
#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#See the License for the specific language governing permissions and
#limitations under the License.
50 \current def  #define current as 50A
[1.0*] \o def       #define o as an (identity) operator for Ohms units
[0.001*] \mo def    #define mo as an operator that converts milliOhms to Ohms
[0.000001*] \uo def #define uo as an operator that converts microOhms to Ohms
[ current * dup "Voltage Drop: " swap cat "V" cat print current * "Power (watts): " swap cat "W" cat print ] \mosfet def
#[ #mosfet calculations (takes as input reistance in Ohms)
#  current *                      #multiply resistance by current
#  dup                            #dup so we have a second copy to calculate power
#  "Voltage Drop: " swap cat print  #print vdrop
#  current *                      #multiply by current again to get power
#  "Power (watts): " swap cat print #print power
#] \mosfet def

#now we can use mosfet with resistance on the top of the stack to run the function
50 mo mosfet
175 uo mosfet

#OR define mosfet to take milliOhms
[ mo current * dup "Voltage Drop: " swap cat "V" cat print current * "Power (watts): " swap cat "W" cat print ] \momosfet def
50 momosfet
175000 momosfet
