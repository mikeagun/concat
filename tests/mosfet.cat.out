Voltage Drop: 2.500000V
Power (watts): 125.000000W
Voltage Drop: 0.008750V
Power (watts): 0.437500W
Voltage Drop: 2.500000V
Power (watts): 125.000000W
Voltage Drop: 8750.000000V
Power (watts): 437500.000000W
